src/ar_storage.erl

```diff
@@ -178,25 +203,39 @@ write_encrypted_block(Hash, B) ->
 			),
 			{error, enospc}
 	end.
--endif.
-
-%% @doc Read a block from disk, given a hash.
-read_block(unavailable, _BHL) -> unavailable;
-read_block(B, _BHL) when is_record(B, block) -> B;
-read_block(Bs, BHL) when is_list(Bs) ->
-	lists:map(fun(B) -> read_block(B, BHL) end, Bs);
-read_block(ID, BHL) ->
-	case ar_block_index:get_block_filename(ID) of
-		unavailable -> unavailable;
-		Filename -> read_block_file(Filename, BHL)
+
+%% @doc Read the block from disk, given a hash or a height.
+read_block(unavailable, _HL) ->
+	unavailable;
+read_block(B, _HL) when is_record(B, block) ->
+	B;
+read_block(Blocks, HL) when is_list(Blocks) ->
+	lists:map(fun(B) -> read_block(B, HL) end, Blocks);
+read_block(Height, HL) when is_integer(Height) ->
+	case Height of
```

Comment:
I know we're not big fans of the erlang if statement, but this is really asking to be one